tests/integration/src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala

```diff
@@ -145,12 +147,14 @@ case class MesosCluster(
 
   def waitForLeader(): Future[String] = async {
     val firstMaster = s"http://${masters.head.ip}:${masters.head.port}"
```

Comment:
Above we start `masters` and `agents` processes. A `Process` has `def isAlive(): Boolean` method. We could call this method here and below in `waitForAgents()` checking for process status. If it's dead, there is no reason to poll `/mesos/state` for the next 5min.

Reply:
The exit happens after twenty seconds though. I think we would be too fast.